### PR TITLE
#2594 Updated I_22, I_23 and I_47 validation rules

### PR DESCRIPTION
**Pull request by MalinaStefaniaStoicanescu**
Change-Id: If4785fb75d2de11389c0eab733a9d55c6a1e735e

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/CapellaElementInDescriptionExistanceCheck.java [hunk 1]
```diff
@@ -63,17 +94,115 @@ public void startElement(String uri, String localName, String qName, Attributes
...
+        isElementExist = true;
+        // empty the value and the id
+        elementValue = new StringBuilder();
+        elementId = "";
+      }
+    }
+
+  }
+
+  public static String getName(EObject object) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/CapellaElementInDescriptionExistanceCheck.java:151]: First, code duplication with CapellaElementInDescriptionNameCheck . Second, look for an existing method that gives the element name, if not exist, contribute in one of the generic helper and use here.

File: core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/DeleteInValidHyperLinkInDescription.java [hunk 2]
```diff
@@ -47,13 +48,34 @@ public DeleteInValidHyperLinkInDescription() {
     _currentElementDescription = null;
   }
 
-  public boolean updateDescription(List<EObject> modelElements) {
+  private String getDescription(EObject object) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/DeleteInValidHyperLinkInDescription.java:51]: Code duplication. Same remark, look for existing helper, if not exist, contribute a new one.

File: core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/I_22_Resolver.java [hunk 3]
```diff
@@ -31,11 +34,25 @@ public class I_22_Resolver extends AbstractCapellaMarkerResolution {
 
   protected Logger _logger = ReportManagerRegistry.getInstance().subscribe(IReportManagerDefaultComponents.VALIDATION);
 
+  public String extractId(String statusMessage) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/I_22_Resolver.java:37]: Move all these extractXXX from statusMessage into 1 helper.

File: core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/I_23_ConvertResolver.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import org.polarsys.capella.common.tools.report.appenders.reportlogview.MarkerViewHelper;
+import org.polarsys.capella.common.tools.report.config.registry.ReportManagerRegistry;
+import org.polarsys.capella.common.tools.report.util.IReportManagerDefaultComponents;
+import org.polarsys.capella.core.validation.ui.ide.quickfix.AbstractCapellaMarkerResolution;
+
+public class I_23_ConvertResolver extends AbstractCapellaMarkerResolution {
+
+  protected Logger _logger = ReportManagerRegistry.getInstance().subscribe(IReportManagerDefaultComponents.VALIDATION);
+
+  public String extractId(String statusMessage) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.platform.sirius.sirius.validation/src/org/polarsys/capella/core/platform/sirius/sirius/validation/ddiagram/I_23_ConvertResolver.java:36]: Code duplication

File: tests/plugins/org.polarsys.capella.test.validation.rules.ju/src/org/polarsys/capella/test/validation/rules/ju/testcases/i/Rule_I_23.java [hunk 5]
```diff
@@ -1,63 +1,117 @@
...
+  /**
+   * @see org.polarsys.capella.test.validation.rules.ju.testcases.ValidationRuleTestCase#getOracleDefinitions()
+   * @generated
+   */
+  protected List<OracleDefinition> getOracleDefinitions() {
+    return Arrays
+        .asList(new ComposedOracleDefinition("e3012a49-86bc-44a1-b91c-7c18f232ac01", 3, Collections.emptyList()));
+  }
+
+  public String extractId(String statusMessage) {
```
Comment (minhtutonthat) [inline tests/plugins/org.polarsys.capella.test.validation.rules.ju/src/org/polarsys/capella/test/validation/rules/ju/testcases/i/Rule_I_23.java:77]: Remove this

File: tests/plugins/org.polarsys.capella.test.validation.rules.ju/src/org/polarsys/capella/test/validation/rules/ju/testcases/i/Rule_I_47_IncorrectImagePath.java [hunk 6]
```diff
@@ -75,6 +86,22 @@ public void test() throws Exception {
     assertEquals("Bad diagnostic children number", 1, children.size());
     assertEquals("Bad diagnostic severity", Diagnostic.ERROR, children.get(0).getSeverity());
     assertTrue("Bad diagnotic message", children.get(0).getMessage().contains("(Image) The absolute path"));
+    List<IMarker> markers = new ArrayList<IMarker>();
```
Comment (minhtutonthat) [inline tests/plugins/org.polarsys.capella.test.validation.rules.ju/src/org/polarsys/capella/test/validation/rules/ju/testcases/i/Rule_I_47_IncorrectImagePath.java:89]: Move this up to one of the abstract classes, like AbstractRulesOnIntegrityTest ?

**Comment by pdulth** [comment]
Already merged

**Pull request closed without merging**